src/python/deepgnn/graph_engine/snark/tests/e2e_test.py

```diff
@@ -19,15 +18,15 @@
 from grpc_health.v1 import health_pb2, health_pb2_grpc
 
 import deepgnn.graph_engine.snark.client as client
-from deepgnn.graph_engine.snark.decoders import DecoderType
+from deepgnn.graph_engine.snark.decoders import LinearDecoder, json_node_to_linear
 import deepgnn.graph_engine.snark.server as server
 import deepgnn.graph_engine.snark.convert as convert
 import deepgnn.graph_engine.snark.dispatcher as dispatcher
 import deepgnn.graph_engine.snark._lib as lib
 
 
-def triangle_graph_json(folder):
-    data = open(os.path.join(folder, "graph.json"), "w+")
+def triangle_graph_linear(folder):
```

Comment:
Please, test linear format along with json and tsv, instead of just replacing json, i.e. format is test parameter.